node/src/neighborhood/mod.rs

```diff
@@ -4235,28 +4352,66 @@ mod tests {
     }
 
     #[test]
-    fn handle_new_public_ip_changes_public_ip_and_nothing_else() {
+    fn handle_new_public_ip_changes_public_ip_and_announces_the_change() {
         init_test_logging();
         let subject_node = make_global_cryptde_node_record(1234, true);
-        let neighbor = make_node_record(1050, true);
-        let mut subject: Neighborhood = neighborhood_from_nodes(&subject_node, Some(&neighbor));
         let new_public_ip = IpAddr::from_str("4.3.2.1").unwrap();
+        let (hopper, _, hopper_recording_arc) = make_recorder();
+        let (subject, one_neighbor, another_neighbor) = {
+            let mut subject: Neighborhood = neighborhood_from_nodes(&subject_node,
+                Some(&make_node_record(7654, true)));
+            subject.gossip_producer_opt = Some(Box::new (GossipProducerReal::new()));
+            subject.hopper_no_lookup_opt = Some (hopper.start().recipient());
```

Comment:
I think we should rename the filed in `Neighborhood` from `hopper_no_lookup_opt` to `hopper_no_lookup_sub_opt`. Will you please?

Reply:
I renamed a couple of others as well.